src/basic/random-util.c

```diff
@@ -93,34 +93,31 @@ int acquire_random_bytes(void *p, size_t n, bool high_quality_required) {
 }
 
 void initialize_srand(void) {
-        static bool srand_called = false;
-        unsigned x;
+        static bool initialized = false;
+        unsigned x = 0;
 #ifdef HAVE_SYS_AUXV_H
-        void *auxv;
+        char *auxv;
 #endif
 
-        if (srand_called)
+        if (initialized)
                 return;
 
+        x ^= (unsigned) now(CLOCK_REALTIME);
+        x ^= (unsigned) gettid();
+
+        initialized = true;
+
 #ifdef HAVE_SYS_AUXV_H
         /* The kernel provides us with 16 bytes of entropy in auxv, so let's
          * try to make use of that to seed the pseudo-random generator. It's
          * better than nothing... */
 
-        auxv = (void*) getauxval(AT_RANDOM);
-        if (auxv) {
-                assert_cc(sizeof(x) < 16);
-                memcpy(&x, auxv, sizeof(x));
-        } else
+        auxv = (char*) getauxval(AT_RANDOM);
+        if (auxv)
+                (void) initstate(x, auxv, 16);
```

Comment:
I mean, I should have checked more carefully, but omg, there's a big buffer for state which is set with a 31 bit seed. What's the point? I'll just drop that commit, it was only for fallback anyway.